### ISSUE TITLE
Define initial GraphQL entrypoint(s)

### ISSUE THREAD

**Issue by alancleary** [legumeinfo/graphql-prototypes#2]
The first use-case we want to accommodate is a Gene Search. Define the GraphQL query type with an entrypoint that will enable such a search. (I included myself on this issue so I can look into how to implement pagination)

**Comment by alancleary** [comment]
@adf-ncgr Regarding pagination, do you have an opinion on numbered vs sequential (cursor)? Numbered is what's currently implemented in the Apollo prototype but that's only because InterMine uses numbered.

Here's an article on the different types of pagination: https://www.apollographql.com/blog/graphql/pagination/understanding-pagination-rest-graphql-and-relay/

**Comment by adf-ncgr** [comment]
I would probably lean towards numbered especially in cases where sorting is applied, but not sure I have a good sense for how it will come into play. Infinite scroll sounds fun. 
Whatever we do, let's avoid: 
"it makes your website seem janky if while navigating sequential pages or scrolling through an infinite scrolling newsfeed I see the same item twice in a row"
@ctcncgr would never let us allow apparent jankiness

**Comment by sammyjava** [comment]
Janky is all about appearances.

**Comment by alancleary** [comment]
> Janky is all about appearances.

Wisdom.

**Comment by adf-ncgr** [comment]
looking back over the pagination discussion you linked, I think it's pretty clear that "drawbacks of page numbering" do not apply to us, since the nature of our information definitely has more of the static flavor. This probably lessens the danger of jank (except when the static data is itself janky).

**Comment by alancleary** [comment]
I agree. I was just curious if you maybe had a use case where you would want a paginated link that was bookmarked or something to not break when new data is added to the data store. I think numbered pages makes the most sense, although we should probably should change the paging parameters to be more verbose since they'll be mixed with type-specific parameters, e.g. `start` -> `page` and `size` -> `pageSize`.
